Makefile

```diff
@@ -49,10 +49,10 @@ else
 endif
 
 pyflakes:
-	find . \( -name _build -o -name var -o -path ./docs -o -path ./mu/contrib -o -path ./utils -o -path ./venv -o -path ./package \) -type d -prune -o -name '*.py' -print0 | $(XARGS) pyflakes
+	pyflakes make.py ./mu/ ./package/ run.py setup.py ./tests/ win_installer.py
```

Comment:
My main concern in this case is that if we had a new file in the root directory it would not be checked unless this line is edited, right?
In that case somebody could add a new file, run the tests, all pass and there would be no reason to not trust the CI results.

Reply:
Quis custodiet ipsos custodes?

I don't think it's the worst thing in the world if we miss running pyflakes on a file. And whatever solution we come up (eg a metapyflakes command which checks which files are run by pyflakes) would fall foul of the situation where decided to add a new checker... :)

Reply:
@carlosperate, on your note about "new file in the root directory", I'll refer to #891 where I propose a few changes to the general repository layout (main motive: repo readability):

* `run.py` should go away as there is (I suppose) no reason for it to exist, any longer.
* `win_installer.py` should be moved under `./package/windows` or something.
* We're left with `setup.py` and `make.py`.
* I see no reason for more `.py` files in the root of the repository.

Nonetheless, my other comment, concerning the use of `flake8` instead of `pycodestyle` and `pyflakes` (which seems to be nearly the same), has the potential to address such "new files" automatically. :)

Reply:
Yes, but something like `pyflakes ./*.py` seems to work to look at all the python files in the root directory without looking into subdirectories. Don't know if it's just luck or valid argument syntax, but if it is valid we could just replace this with:

```
pyflakes  ./*.py ./mu/ ./package/ ./tests/
```

Reply:
> Yes, but something like `pyflakes ./*.py` seems to work to look at all the python files in the root directory without looking into subdirectories. Don't know if it's just luck or valid argument syntax, but if it is valid we could just replace this with:

...please see my latest note / commits (down there!) where `flake8` makes everybody's life simpler, while handling all the cases we've been discussing. :)

Reply:
Sorry, for some reason when I typed my reply my browser didn't show me your reply to this thread, the response on the main thread, nor the new commits (I assume it showed me an old cached version from last night).

I'll have to read all the other issues and PRs some other time (probably during the weekend, there is a lot to digest), but as a quick question, what is the alternative to `run.py`? I still use as my normal entry point to running Mu from source.

Reply:
> [...] but as a quick question, what is the alternative to `run.py`?

```
$ mu-editor
```

or, if you want to go all Pythonic,

```
$ python -m mu
```

Reply:
`mu-editor` depends on having installed the package in editable mode, which I am always a bit sceptical about, since I've run into bugs in this area within pip, pipenv and poetry (all fixed eventually, but now I am always second guessing if I'm running the version of the code I think I am running).
`python -m mu` depends on running that command from the project root or it will run the installed version instead of the local one, which is fine, but I guess I liked how explicit `python run.py` was :)

Reply:
> `mu-editor` depends on having installed the package in editable mode, which I am always a bit sceptical about, since I've run into bugs in this area within pip, pipenv and poetry (all fixed eventually, but now I am always second guessing if I'm running the version of the code I think I am running).

I think you may be misunderstanding it: `mu-editor` is a `setuptools` [entry point](https://setuptools.readthedocs.io/en/latest/setuptools.html#automatic-script-creation) that will be created on all platforms after installation regardless of it being i) from source, in editable mode, ii) from source in "normal mode" with, say, `pip install .`, or iii) from a local or PyPI sourced wheel. Give it a try! :)

> `python -m mu` depends on running that command from the project root or it will run the installed version instead of the local one, which is fine, but I guess I liked how explicit `python run.py` was :)

I guess I understand your perspective but it is very important to run the "installed" version for many reasons (example, you want to ensure that resources such as, say, icons, are properly packaged/installed). Developers tend to run the "installed" versions which "matches" the source version (thus working with editable installs).

For these reasons (which deserve a more complete discussion/explanation) I also propose moving to an `src`-based layout in #891 -- but no hurries there, and happy to discuss. :)

PS: I mentioned `python -m mu` for completeness. We could make it "go away", I guess, and provide a single way of running Mu. :)

Reply:
I should have clarified that I meant my comment was only from the perspective of running Mu from source as a developer that plans to modify the source code and run that version of Mu (which is why I was mentioning installing in editable mode).

Of course users **should not** be running it this way, and they have different installation instructions for that. And there are other reasons to run the installed version, which all can happily coexists with the `run.py` file. But if it has to go it has to go, I'll just have to be careful when running `python -m mu`.